dubbo-registry/dubbo-registry-zookeeper/src/main/java/org/apache/dubbo/registry/zookeeper/ZookeeperRegistry.java

```diff
@@ -126,6 +151,41 @@ public void doUnregister(URL url) {
         }
     }
 
+    private void checkZkProvider(){

```

Comment:
One second thought, I am wondering why the following code in org.apache.dubbo.registry.zookeeper.ZookeeperRegistry doesn't work:

```java
        zkClient.addStateListener(new StateListener() {
            @Override
            public void stateChanged(int state) {
                if (state == RECONNECTED) {
                    try {
                        recover();
                    } catch (Exception e) {
                        logger.error(e.getMessage(), e);
                    }
                }
            }
        });
```

Would you mind to take a look?